src/main/kotlin/se/uulm/snowballr/backend/repository/UserTableRepo.kt

```diff
@@ -273,6 +293,43 @@ class UserTableRepo(
         }
     }
 
+    override suspend fun clearSoftDeletedUsers(thresholdDate: OffsetDateTime) = db.query {
+        // Clear (user-)criteria for each user to be cleared
+        val usersToBeCleared = UserTable.selectAll()
+            .where {
+                (UserTable.status eq UserStatus.USER_STATUS_DELETED).and(UserTable.deletedAt lessEq thresholdDate)
+            }
+            .map { it[UserTable.id].value }
+
+        usersToBeCleared.forEach { userId ->
+            CriterionTable.deleteWhere {
+                (CriterionTable.createdBy eq userId).and(CriterionTable.projectId.isNull())
+            }
+        }
+
+        val clearedUsers = UserTable.update(
+            {
+                UserTable.id inList usersToBeCleared
+            },
+        ) {
+            it[email] = ""
+            it[firstName] = ""
+            it[lastName] = ""
+            it[passwordHash] = ""
+            it[role] = USER_ROLE_UNSPECIFIED
+            it[status] = USER_STATUS_UNSPECIFIED
+
+            it[criteriaIds] = emptyList() // Criteria should be deleted before the user is deleted!
+
+            it[fetchers] = emptyMap()
+            it[snowballingType] = SNOWBALLING_TYPE_UNSPECIFIED
+
+            it[modifiedAt] = OffsetDateTime.now()
```

Comment:
```suggestion
            it[deletedAt] = OffsetDateTime.now()
```
I would propose to change `deletedAt` here, since we logically delete the user and it should not affect the hard-delete if we change `deletedAt`.

Reply:
No. The `deletedAt` field is set when the user is soft-deleted. This value is then used for this exact method to evaluate if the user is to be cleared. This value is also used in the hard-delete function to again validate that the user is ready to be cleared because threshold date is reached. Therefore we modify the `modifedAt` field to mark the clearing and keep the `deletedAt` field as is.

Reply:
Ok, I get your point here. I'm not totally happy with `modifiedat`, since this happens after the deletion and is more a cleaning than a modification in a logical sense, but I get your intention and have no better idea for now. Resolve it by your own.